fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java

```diff
@@ -90,6 +98,190 @@
  */
 public class MaterializedViewUtils {
 
+    public static final String MIN = "min";
+    public static final String MAX = "max";
+    public static final String SUM = "sum";
+
+    /**
+     * paimon mv agg keys and functions
+     */
+    public static class MVAggInfo {
+        public List<String> keys = new ArrayList<>();
+        public Map<String, String> aggFunctions = new HashMap<>();
+    }
+
+    private static class MVAnalysis implements GeneratedPlanPatterns {
+        public static MVAnalysis instance = new MVAnalysis();
+        private static final ImmutableSet<String> ALLOW_MERGE_AGGREGATE_FUNCTIONS =
+                ImmutableSet.of(MIN, MAX, SUM);
+
+        /**
+         * Extract aggregation keys and aggregation functions according to the execution plan
+         */
+        public MVAggInfo checkAndGetAggInfo(Plan plan) {
+            if (plan instanceof LogicalResultSink) {
+                plan = plan.child(0);
+            }
+            boolean isAggMode = logicalAggregate(logicalProject(logicalFileScan()))
+                    .getPattern().matchPlanTree(plan);
+            if (!isAggMode) {
+                throw new RuntimeException(
+                        "mv query must be agg mode and table must a paimon table: " + plan);
+            }
+            LogicalAggregate<? extends Plan> aggregate = (LogicalAggregate) plan;
+            LogicalProject<? extends Plan> project = (LogicalProject) plan.child(0);
+            LogicalFileScan fileScan = (LogicalFileScan) project.child(0);
+
+            if (!(fileScan.getTable() instanceof PaimonExternalTable)) {
+                throw new RuntimeException("table " + fileScan.getTable() + " not support incremental mv");
```

Comment:
How about using `NereidsException`? Same for other `RuntimeException` in this class
And change the exception message: "Only Paimon table support incremental mv"